File: Using Java/SortStack.java
import java.util.Stack;

public class SortStack {

    static void sortStack(Stack<Integer> st){

        if(st.empty()){
            return;
        }

        int element = st.pop();

        sortStack(st);
        insertInSortedOrder(st, element);
    }

    static void insertInSortedOrder(Stack<Integer> st, int element){

        if(st.empty()){
            st.push(element);
            return;
        }

        if(st.peek() > element){
            st.push(element); 
        }

        else {

            int top = st.pop();
            insertInSortedOrder(st, element);
            st.push(top);
        }

    }


    public static void main(String[] args) {

        Stack<Integer> st = new Stack<>();

        st.push(9);
        st.push(5);
        st.push(14);
        st.push(1);
        st.push(20);

        System.out.println(st);

        sortStack(st);
        System.out.println(st);
    }
}

File: data structures questions/sortStack.cpp
#include<bits/stdc++.h>
using namespace std;

#include <bits/stdc++.h> 

void insertInSortedOrder(stack<int>& st, int element){
    
    if(st.empty() || st.top() <= element){
        st.push(element);
        return;
    }
    
    int e = st.top();
    st.pop();
    
    insertInSortedOrder(st, element);
    st.push(e);
    
}
void solve(stack<int>& st){
    
    if(st.empty()){
        return;
    }
    
    int element = st.top();
    st.pop();
    
    solve(st);
    
    insertInSortedOrder(st, element);
}

int main(){

    stack<int> st;
    st.push(5);
    st.push(3);
    st.push(4);
    st.push(1);

    solve(st);

    while(!st.empty()){
        cout << st.top() << endl;
        st.pop();

    }

    return 0;
}